File: CvTracker.Client/models/UserProfile.ts
import type { UserTechnology } from './UserSkill'

export interface UserProfile {
  id: number
  firstName: string | null
  lastName: string | null
  location: string | null
  linkedInUrl: string | null
  gitHubUrl: string | null
  websiteUrl: string | null
  avatarUrl: string | null
  resumeFileName: string | null
  resumeUrl: string | null
  skills: UserTechnology[]
}

export interface UpdateUserProfileRequest {
  firstName: string | null
  lastName: string | null
  location: string | null
  linkedInUrl: string | null
  gitHubUrl: string | null
  websiteUrl: string | null
}

File: CvTracker.Api/Controllers/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;

public class UserProfile
{
    public int Id { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Location { get; set; }
    [Url] public string? LinkedInUrl { get; set; }
    [Url] public string? GitHubUrl { get; set; }
    [Url] public string? WebsiteUrl { get; set; }
    public string? AvatarFileName { get; set; }
    public string? ResumeFileName { get; set; }
}
